pkg/report/testdata/starnix/report/0

```diff
@@ -0,0 +1,84 @@
+TITLE: starnix kernel panic: explicit panic in src/starnix/kernel/runner/container.rs: failed to start container.: errno NUM, d
+CORRUPTED: Y
```

Comment:
Corrupted reports are thrown away. It's good to have some for testing, but it's more important to have non-corrupted test samples that are parsed correctly.
Please add some that are parsed as non-corrupted.

Reply:
I think (please correct me if I'm wrong) that this particular report should be parsed as non-corrupted; I just hadn't added starnix-specific stack frame patterns yet in the first version of this PR. I've added some patterns that I think are sufficient for Rust panics in the most recent push, and this report is no longer marked as corrupted.

Reply:
I have a couple of questions related to stack frame parsing that I'll add on here:

1. In order to get this report classified as non-corrupted, it was enough to add the `rustBacktrace` pattern to the stack params' `stackStartRes` to describe what the beginning of a stack trace should look like. In addition, I added some regexes (as `starnixFramePatterns`) matching individual frames. I checked (with printf debugging) that each line of the stack trace in `pkg/report/testdata/starnix/report/0` matches one of the `starnixFramePatterns` regexes, but I'm not sure what the report parser should do with these matches, if anything. It seems that they don't affect the report. I'm guessing we'll want to add some functionality to the fuchsia report impl to process the matches; are there e.g. good examples of tests that we should try to pass?

2. Currently, with our local fuzzer running this commit, the reports include a ton of console output. This might be worse for starnix than for other targets since the underlying fuchsia system can keep emitting logs after starnix crashes. It seems to me that we want to trim the console output to include only the relevant stack trace. For example, in the test report here, we don't want to include the last line in the report, since it doesn't match any of the `starnixFramePatterns` regexes. Is that correct? Have you had to trim console output for other types of reports?

Reply:
> but I'm not sure what the report parser should do with these matches, if anything. It seems that they don't affect the report.

If there are no frames at all, we declare the report as corrupted. Also, especially for Linux, we rely quite heavily on frames to determine the title for each report: we skip common function names to reach more specific locations in the backtrace (you will see a lot of stuff if you grep `linux.go` for `skip`). 

https://github.com/google/syzkaller/blob/d902085f716b92f42bb867eca2975c7da711b78d/pkg/report/report.go#L512-L518

>  I'm guessing we'll want to add some functionality to the fuchsia report impl to process the matches; are there e.g. good examples of tests that we should try to pass?

It depends on your needs. For Linux, we've added report symbolization (which seems to be already done by rust in your case?), opcode decompilation (which also does not seem relevant) + the code to better parse task hung reports (where the simple "traverse the stack trace and skip commonly used functions" does not suffice). The rest are just tons of `oops` structure initializations for the specific kinds of Linux crash reports.

The best you can do is to look at the reports generated by your local instance. Once titles and the reports seem adequate, you won't have to do anything else.

> Have you had to trim console output for other types of reports?

AFAIK for Linux we usually don't have this problem -- it prepends context prefixes to the serial output  (so we can remove all other output) + crash reports are mostly short.

In your case, it may make sense to look at how it's done for gVisor:

https://github.com/google/syzkaller/blob/d902085f716b92f42bb867eca2975c7da711b78d/pkg/report/gvisor.go#L52-L56

Reply:
Belated thanks for this comment, it was helpful! I added a trimming function that is specific to Rust panics. It runs as a second pass after the main `shortenReport` and tries to extract the stack trace without too many unrelated lines. Unfortunately, there are often groups of unrelated log lines interspersed with the interesting lines. This version errs on the side of capturing some extra lines; the main goal is just to trim the large number of unrelated lines that usually follow the end of the stack trace. 

I added the expected report output (after `REPORT:`) to the unit tests, so the report format is now tested in addition to the title and corrupted/non-corrupted status.

Please let me know what you think -- I'm sure this will need some more tuning in the future, but for the kinds of crashes we've seen so far, it seems like a reasonable start.